servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/FromInputStreamPublisher.java

```diff
@@ -306,4 +291,32 @@ private boolean trySetTerminalSent() {
             return true;
         }
     }
+
+    static final class ToByteArrayMapper implements ByteArrayMapper<byte[]> {
+
+        static final ByteArrayMapper<byte[]> DEFAULT_TO_BYTE_ARRAY_MAPPER =
+                new ToByteArrayMapper(DEFAULT_MAX_BUFFER_SIZE);
+
+        private final int maxBufferSize;
+
+        ToByteArrayMapper(final int maxBufferSize) {
+            this.maxBufferSize = ensurePositive(maxBufferSize, "maxBufferSize");
+        }
+
+        @Override
+        public byte[] map(final byte[] buffer, final int offset, final int length) {
+            if (offset == 0 && length == buffer.length) {
+                return buffer;
+            } else {
+                final byte[] partial = new byte[length];
```

Comment:
since we are also allocating a new array like with the old code, how does that improve the GC performance in the default cases? (not the ones with a custom ByteArrayMapper)

Reply:
This exists only for backward compatibility when users already depend on T as byte-array and need to migrate from deprecated methods. All internal use-cases were changed to use `BufferAllocator.wrap` that helps to avoid this path